homeassistant/components/binary_sensor/netatmo.py

```diff
@@ -130,30 +169,54 @@ def update(self):
         self._data.update()
         self._data.update_event()
 
-        if self._sensor_name == "Someone known":
-            self._state =\
-                    self._data.welcomedata.someoneKnownSeen(self._home,
+        if self._cameratype == "NACamera":
+            if self._sensor_name == "Someone known":
+                self._state =\
+                    self._data.camera_data.someoneKnownSeen(self._home,
                                                             self._camera_name,
                                                             self._timeout*60)
-        elif self._sensor_name == "Someone unknown":
-            self._state =\
-                self._data.welcomedata.someoneUnknownSeen(self._home,
+            elif self._sensor_name == "Someone unknown":
+                self._state =\
+                    self._data.camera_data.someoneUnknownSeen(
+                        self._home, self._camera_name, self._timeout*60)
+            elif self._sensor_name == "Motion":
+                self._state =\
+                    self._data.camera_data.motionDetected(self._home,
                                                           self._camera_name,
                                                           self._timeout*60)
-        elif self._sensor_name == "Motion":
-            self._state =\
-                self._data.welcomedata.motionDetected(self._home,
-                                                      self._camera_name,
-                                                      self._timeout*60)
+            else:
+                return None
+        elif self._cameratype == "NOC":
+            if self._sensor_name == "Outdoor motion":
+                self._state =\
+                    self._data.camera_data.outdoormotionDetected(
+                        self._home, self._camera_name, self._offset)
+            elif self._sensor_name == "Outdoor human":
+                self._state =\
+                    self._data.camera_data.humanDetected(self._home,
+                                                         self._camera_name,
+                                                         self._offset)
+            elif self._sensor_name == "Outdoor animal":
+                self._state =\
+                    self._data.camera_data.animalDetected(self._home,
+                                                          self._camera_name,
+                                                          self._offset)
+            elif self._sensor_name == "Outdoor vehicle":
+                self._state =\
+                    self._data.camera_data.carDetected(self._home,
+                                                       self._camera_name,
+                                                       self._offset)
+            else:
+                return None
         elif self._sensor_name == "Tag Vibration":
```

Comment:
Tag are linked to a Welcome Camera (NACamera), I think 'self._sensor_name == "Tag Vibration"' and 'self._sensor_name == "Tag Open"' should be under ' if self._cameratype == "NACamera"'